server/src/main/java/org/elasticsearch/search/fetch/FetchPhase.java

```diff
@@ -342,12 +348,15 @@ private SearchHit createNestedSearchHit(SearchContext context,
             XContentType contentType = tuple.v1();
             context.lookup().source().setSourceContentType(contentType);
         }
-
-        Map<String, DocumentField> metaFields = new HashMap<>(),
-            documentFields = new HashMap<>();
-        SearchHit.splitFieldsByMetadata(searchFields, documentFields, metaFields);
-
-        return new SearchHit(nestedTopDocId, id, nestedIdentity, documentFields, metaFields);
+        if ((searchFields == null) || searchFields.size() == 0) {
+            return new SearchHit(nestedTopDocId, id, nestedIdentity, emptyMap(), emptyMap());
+        } else {
+            Map<String, DocumentField> metaFields = new HashMap<>();
+            Map<String, DocumentField> documentFields = new HashMap<>();
+            searchFields.forEach((fieldName, docField) ->
```

Comment:
just a question: the searchFields seem to contain both types of DocumentFields here (they are thrown together earlier). Would if be feasible somehow to not lump them together in the first place, then we wouldn't need to split again here? Or does this make other things to complicated?